ComSemApp/templates/ComSemApp/corpus/corpus_search.html

```diff
@@ -256,218 +172,154 @@ <h4 class="m-t-0 header-title">
-				return false;
-
-			var toAdd = "<tr><td>offset</td><td>" + offsetVal + "</td><td><button class='btn btn-sm btn-outline-danger float-right' type='button' name='DeleteRow'><i class='ti-trash'></i></button></td><input hidden type='text' value='' name='offset[" + numOffsets + "]'></tr>";
-
-			numOffsets = numOffsets + 1;
-
-			appendCriterion('offset', offsetVal, null)
-
-			$("#sortParams table").append(toAdd).show();
-		});
-
-		// clear selector
-		$("button[name='ClearSelector']").click(function() {
-			$("#DynamicField").empty();
-		});
-
-		//clear query Builder
-		$('#clearQueryBuilder').click(function(){
-			$('#sortParams table').hide().find('tbody').html('');
-			searchCriteria = [];
-			$('#searchCriteria').val(JSON.stringify(searchCriteria));
-		});
-
-
-		// submit selector
-		$("button[name='SubmitSelector']").click(function() {
-
-			$(this).prop("disabled",true);
-			var type = $('#search_type').val()
-			var id_list = [];
-			var val = $("input[name='val']").val();
-
-			$("input[name^='selector_checked']").each(function() {
-				if ($(this).is(':checked')) {
-					id_list.push($(this).val());
-				}
+	/*
+	If the user marks the 'select all' option in a POS check list, each element
+	is toggled so that checkAll is selected, each element is also selected.
+	If checkAll is deselected, each element is also deselected, or left alone
+	if they are already deselected.
+	*/
+	function selectAll(form) {
+		if ($("input[name^='" + form + "_checkAll']").is(':checked')) {
+			$("input[name^='" + form + "_selector_checked']").each(function() {
+				$(this).prop('checked', true);
 			});
-
-			buildSearchQueryRow(type, val, id_list);
-
-		});
-
-	});
-
-
-
-	$(document).on("click", "button[name='DeleteRow']", function() {
-
-		// delete the criteria from the search criteria array, put new json string in input field
-		var index = $(this).parents('tr').index();
-		searchCriteria.splice(index, 1);
-		$('#searchCriteria').val(JSON.stringify(searchCriteria));
-
-		// remove the row from the document
-        $(this).closest('tr').remove();
-    });
-
-
-
-	$('input[name="new_word"]').keypress(function(e){
-		if (e.which == 13) return false;
-	});
-
-	$(".addToSearch").on("click", function(e){
-		e.preventDefault();
-
-		var output = $(this).attr('output'); // refine the search in the second box or go directly to the query builder?
-		var type = $('#new_word').val() ? 'word' : 'tag'; // is this a word or tag? if word input has content, it's a word
-		var val = '';
-
-		$(this).prop("disabled",true);
-		$("#DynamicField").empty();
-
-		if(type == 'word'){
-			val = $("input[name='new_word']").val();
 		} else {
-			val = $("select[name='new_tag']").val();
+			$("input[name^='" + form + "_selector_checked']").each(function() {
+				$(this).prop('checked', false);
+			});
 		}
+	}
 
-
-		if(output == "html"){
-			$("#LoadingInfo").append("<strong>loading...</strong>");
+	/*
+	This function returns as a string the value in an HTML input above.
+	*/
```

Comment:
+1 for all the comments. It's way clearer than it was before what's going on here.